File: GXpert/GXpert.Web/Modules/ServerTypes/Schools/SchoolTimeTableColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ESchoolTimeTableType } from "../Web/Enums.ESchoolTimeTableType";
import { SchoolTimeTableRow } from "./SchoolTimeTableRow";

export interface SchoolTimeTableColumns {
    Id: Column<SchoolTimeTableRow>;
    Date: Column<SchoolTimeTableRow>;
    StartTime: Column<SchoolTimeTableRow>;
    EndTime: Column<SchoolTimeTableRow>;
    PeriodIndex: Column<SchoolTimeTableRow>;
    SchoolClassDivision: Column<SchoolTimeTableRow>;
    TeacherPrn: Column<SchoolTimeTableRow>;
    EType: Column<SchoolTimeTableRow>;
}

export class SchoolTimeTableColumns extends ColumnsBase<SchoolTimeTableRow> {
    static readonly columnsKey = 'Schools.SchoolTimeTable';
    static readonly Fields = fieldsProxy<SchoolTimeTableColumns>();
}

[ESchoolTimeTableType]; // referenced types

File: GXpert/GXpert.Web/Modules/Schools/SchoolTimeTable/SchoolTimeTableColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Schools.Columns;

[ColumnsScript("Schools.SchoolTimeTable")]
[BasedOnRow(typeof(SchoolTimeTableRow), CheckNames = true)]
public class SchoolTimeTableColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public DateTime Date { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public int PeriodIndex { get; set; }
    public string SchoolClassDivision { get; set; }
    public string TeacherPrn { get; set; }
    [EditLink]
    public short EType { get; set; }
}